src/lib/device-online-state/index.ts

```diff
@@ -0,0 +1,280 @@
+import * as Bluebird from 'bluebird';
+import { EventEmitter } from 'events';
+import { DEFAULT_SUPERVISOR_POLL_INTERVAL } from '../env-vars';
+import { once, noop } from 'lodash';
+import { resinApi, root } from '../../platform';
+import { captureException } from '../../platform/errors';
+import {
+	createPromisifedRedisClient,
+	PromisifedRedisClient,
+} from '../redis-promise';
+import * as RedisSMQ from 'rsmq-promise';
+
+export class PollIntervalNotFoundError extends Error {}
+export const getPollIntervalForDevice = (uuid: string): Bluebird<number> => {
+	return resinApi
+		.get({
+			resource: 'device_config_variable',
+			passthrough: {
+				req: root,
+			},
+			options: {
+				$select: 'value',
+				$expand: {
+					device: {
+						$filter: { uuid },
+					},
+				},
+				$filter: {
+					name: 'RESIN_SUPERVISOR_POLL_INTERVAL',
+				},
+			},
+		})
+		.then(([pollInterval]: Array<{ value: number }>) => {
+			if (!pollInterval) {
+				throw new PollIntervalNotFoundError();
+			}
+
+			return pollInterval.value;
+		})
+		.catch(PollIntervalNotFoundError, () => {
+			// query to determine the poll interval for this device...
+			return resinApi
+				.get({
+					resource: 'application_config_variable',
+					passthrough: {
+						req: root,
+					},
+					options: {
+						$select: 'value',
+						$top: 1,
+						$filter: {
+							application: {
+								$any: {
+									$alias: 'a',
+									$expr: {
+										a: {
+											owns__device: {
+												$any: {
+													$alias: 'd',
+													$expr: {
+														d: {
+															uuid: uuid,
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+							name: 'RESIN_SUPERVISOR_POLL_INTERVAL',
+						},
+					},
+				})
+				.then(([pollInterval]: Array<{ value: number }>) => {
+					if (pollInterval) {
+						return pollInterval.value;
+					}
+
+					return DEFAULT_SUPERVISOR_POLL_INTERVAL;
+				});
+		});
+};
+
+export const getDefaultManager = once(() => {
```

Comment:
There is no other manager you can get right?  I think it'd be nicer to do
```suggestion
export const manager = new DeviceOnlineStateManager();
```
and simplify call sites a little (name can be different)

Reply:
Done